File: tools/bench/answers/go_sonnet/s3/04_binary.go
package main

import "fmt"

func toBinary(n int) string {
	if n == 0 {
		return "0"
	}
	bits := ""
	for n > 0 {
		if n%2 == 1 {
			bits = "1" + bits
		} else {
			bits = "0" + bits
		}
		n /= 2
	}
	return bits
}

func main() {
	fmt.Println(toBinary(13))
	fmt.Println(toBinary(64))
}

File: tools/bench/answers/rust_sonnet/s2/04_binary.rs
fn to_binary(mut n: u64) -> String {
    if n == 0 {
        return "0".to_string();
    }
    let mut bits = Vec::new();
    while n > 0 {
        bits.push((n % 2).to_string());
        n /= 2;
    }
    bits.reverse();
    bits.join("")
}

fn main() {
    println!("{}", to_binary(13));
    println!("{}", to_binary(64));
}
